src/hooks/useLanguage.js

```diff
@@ -0,0 +1,162 @@
+import { useState, useMemo, useCallback } from "react";
+
+import _ from "lodash";
+
+import * as Yup from "yup";
+import config from "~/constants/endpoints.json";
+
+const currentConfig = import.meta.env.MODE === "development" ? config.test : config.prod;
+
+const langConfigInitial = {
+  lang_id: "",
+  lang_name: "",
+  tenant_id: ""
+};
+const schema = Yup.object().shape({
+  lang_id: Yup.string().required("lang Id is required"),
+  lang_name: Yup.string().required("Lang Name is required"),
+  tenant_id: Yup.string().required("tenant Id text is required ")
+});
+
+const useLanguage = () => {
+  const [langConfig, setLangConfig] = useState(langConfigInitial);
+  const [errors, setErrors] = useState({});
+  const [pageDialogTitle, setPageDialogTitle] = useState("");
+  const [modalOpenAdd, setModalOpenAdd] = useState(false);
+
+  const validateField = useCallback(
+    (name, value) => {
+      schema
+        .validateAt(name, { [name]: value })
+        .then(() => {
+          setErrors((prevErrors) => {
+            return {
+              ...prevErrors,
+              [name]: ""
+            };
+          });
+        })
+        .catch((error) => {
+          setErrors((prevErrors) => {
+            return {
+              ...prevErrors,
+              [name]: error.message
+            };
+          });
+        });
+    },
+    [schema]
+  );
+  const validateObject = useCallback(
+    (formData) => {
+      schema
+        .validate(formData)
+        .then(() => {
+          setErrors({});
+        })
+        .catch((error) => {
+          setErrors((prevErrors) => {
+            return {
+              ...prevErrors,
+              ...error
+            };
+          });
+        });
+    },
+    [schema]
+  );
+
+  const handleChange = useCallback(
+    (event) => {
+      const { name, value } = event.target;
+      setLangConfig((prevState) => {
+        return {
+          ...prevState,
+          [name]: value
+        };
+      });
+      validateField(name, value);
+    },
+    [validateField, setLangConfig]
+  );
+
+  const populateLanguageForm = (row) => {
+    const { lang_id, lang_name, tenant_id } = row.original;
+    setLangConfig((prevState) => {
```

Comment:
please let us change  this  data should come from API , there is insufficient  or inconsistent data ,
`const populatePageConfigForm = useCallback(
    (row) => {
      const { id } = row.original;
      axiosPrivate
        .get(`/api/protected/${currentConfig.pageConfig}`, {
          params: {
            id
          }
        })
        .then(({ data }) => {
          const pageConfigTemp = {
            id: data.id,
            pageType: data.page_type,
            parent: data.parent,
            headerText: data.header_text,
            language: data.language,
            imageLink: data.img_link,
            headerImage: data.header_image,
            orderNumber: data.order_number,
            description: data.description,
            name: data.name
          };
          setPageConfig((prevPageConfig) => {
            return {
              ...prevPageConfig,
              ...pageConfigTemp
            };
          });
          handleAddModalOpen("Update Page Config");
        })
        .catch((error) => {
          console.log("Error : >> ", error);
        });
    },
    [handleAddModalOpen]
  );`

Reply:
okay dave but why do we need most of the data here like the "parent",headertext","language","imageLink"....etc i thought we dont need those in the language section